node/src/components/consensus/era_supervisor.rs

```diff
@@ -201,11 +218,14 @@ where
         }
     }
 
+    /// The booking block for era N is the last block of era N - AUCTION_DELAY - 1
+    /// To find it, we get the start height of era N - AUCTION_DELAY and subtract 1.
+    /// We make sure not to go below the last upgrade activation point, because we will not have
+    /// any eras from before that.
```

Comment:
I am not sure this sentence is 100% correct. We _will_ go below the last upgrade b/c we subtract 1 from it to get the height of a switch block just before the upgrade.

Reply:
But not in the eras, just in the block height. I'll clarify.